src/efsw/FileSystem.cpp

```diff
@@ -91,13 +92,29 @@ std::string FileSystem::precomposeFileName( const std::string& name ) {
 
 	CFStringNormalize( cfMutable, kCFStringNormalizationFormC );
 
-	char c_str[255 + 1];
-	CFStringGetCString( cfMutable, c_str, sizeof( c_str ) - 1, kCFStringEncodingUTF8 );
-
-	CFRelease( cfStringRef );
-	CFRelease( cfMutable );
+	const char* c_str = CFStringGetCStringPtr( cfMutable, kCFStringEncodingUTF8 );
+	if ( c_str != NULL ) {
+		std::string result( c_str );
+		CFRelease( cfStringRef );
+		CFRelease( cfMutable );
+		return result;
+	}
+	CFIndex length = CFStringGetLength( cfMutable );
+	CFIndex maxSize = CFStringGetMaximumSizeForEncoding( length, kCFStringEncodingUTF8 );
+	if ( maxSize == kCFNotFound ) {
+		CFRelease( cfStringRef );
+		CFRelease( cfMutable );
+		return std::string();
+	}
 
-	return std::string( c_str );
+	std::vector<char> c_buf( maxSize + 1 );
```

Comment:
Here you can just use an `std::string` and avoid a new allocation later in a new `std::string` (unless I'm missing something).

Something like:
```c++
	std::string buf;
	c_buf.resize(maxSize); // You already have the UTF-8 length
	if (CFStringGetCString( cfMutable, buf.data(), buf.size(), kCFStringEncodingUTF8 )) {
		result.resize(strlen(buf.c_str())); // If the was any discrepancy just resize it
		CFRelease( cfStringRef );
		CFRelease( cfMutable );
	} else {
		buf.clear();
	}
	return buf;
```

Reply:
Then you can simply do this:
```c++
&buf[0]
```

Reply:
Fixed.